astropy/modeling/tests/test_astro_sherpa.py

```diff
@@ -0,0 +1,242 @@
+Module to test fitting routines
+"""
+
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+
+
+import numpy as np
+
+from numpy.testing.utils import assert_allclose
+from ...tests.helper import pytest
+
+try:
+    from sherpa.data import DataSimulFit
+    HAS_SHERPA = True
+except ImportError:
+    HAS_SHERPA = False
+
+
+from astropy.modeling.astro_sherpa import SherpaFitter, make_datasets, _astropy_to_sherpa_model
+from sherpa.stats import Chi2
+from astropy.modeling.models import Gaussian1D, Gaussian2D
+
+_RANDOM_SEED = 0x1337
+np.random.seed(_RANDOM_SEED)
+
+
+class TestSherpaFitter(object):
+
+    def setup_class(self):
+        # make data and models to use later!
+        err = 0.1
+
+        self.x1 = np.arange(1, 10, .1)
+        self.dx1 = np.ones(self.x1.shape) * (.1 / 2.)
+        self.x2 = np.arange(1, 10, .05)
+        self.dx2 = np.ones(self.x2.shape) * (.05 / 2.)
+
+        self.model1d = Gaussian1D(mean=5, amplitude=10, stddev=0.8)
+        self.model1d_2 = Gaussian1D(mean=4, amplitude=5, stddev=0.2)
+
+        self.tmodel1d = self.model1d.copy()
+        self.tmodel1d_2 = self.model1d_2.copy()
+
+        self.y1 = self.model1d(self.x1)
+        self.y1 += err * np.random.uniform(-1., 1., size=self.y1.size)
+        self.dy1 = err * np.random.uniform(0.5, 1., size=self.y1.size)
+
+        self.y2 = self.model1d_2(self.x2)
+        self.y2 += err * np.random.uniform(-1., 1., size=self.y2.size)
+        self.dy2 = err * np.random.uniform(0.5, 1., size=self.y2.size)
+
+        self.model1d.mean = 4
+        self.model1d.amplitude = 6
+        self.model1d.stddev = 0.5
+
+        self.model1d_2.mean = 5
+        self.model1d_2.amplitude = 10
+        self.model1d_2.stddev = 0.3
+
+        self.xx2, self.xx1 = np.mgrid[1:10:.1, 1:10:.05]
+        self.shape = self.xx2.shape
+        self.xx1 = self.xx1.flatten()
+        self.xx2 = self.xx2.flatten()
+
+        self.model2d = Gaussian2D(amplitude=10, x_mean=5, y_mean=6, x_stddev=0.8, y_stddev=1.1)
+        self.model2d.theta.fixed = True
+
+        self.yy = self.model2d(self.xx1, self.xx2)
+        self.dxx1 = err * np.random.uniform(0.5, 1., size=self.xx1.size)
+        self.dxx2 = err * np.random.uniform(0.5, 1., size=self.xx2.size)
+        self.dyy = err * np.random.uniform(0.5, 1., size=self.yy.size)
+
+        self.tmodel2d = self.model2d.copy()
+        self.model2d.amplitude = 5
+        self.model2d.x_mean = 6
+        self.model2d.y_mean = 5
+        self.model2d.x_stddev = 0.2
+        self.model2d.y_stddev = 0.7
+
+        # to stop stddev going negitive and getting div by zero error
+        self.model1d.stddev.min = 1e-99
+        self.model1d_2.stddev.min = 1e-99
+        self.model2d.x_stddev.min = 1e-99
+        self.model2d.y_stddev.min = 1e-99
+
+        self.fitter = SherpaFitter(statistic=Chi2())
+
+    def test_make_datasets_single(self):
+        data_noerr = make_datasets(1, x=self.x1, y=self.y1)
```

Comment:
Even in tests it's helpful to have a short 1-2 line sentence of what's being tested and why.